src/analysis/retail/warrior/arms/normalizers/FinishingBlowsResourceChange.ts

```diff
@@ -0,0 +1,28 @@
+import SPELLS from 'common/SPELLS/warrior';
+import TALENTS from 'common/TALENTS/warrior';
+import { AnyEvent, EventType } from 'parser/core/Events';
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+
+/**
+ * With the change in War Within pre-patch where Strength of Arms no longer gives 8 rage
+ * when casting Overpower on a target below 35% health, the log still logs with the Strength of Arms
+ * ability, so we just replace it with the Finishing Blows to get a better attribution.
+ */
+export default class FinishingBlowsResourceChange extends EventsNormalizer {
```

Comment:
i don't necessarily agree with this being a normalizer. normally this is something that you can group later without too much difficulty. does this using the "wrong" ability have other bad downstream effects?

Reply:
Do you disagree with the entire "fixing" of rage happening in normalizers, or specifically of this rewriting one spell to another?

My starting point was looking at the [protection RageGenerationNormalizer](https://github.com/WoWAnalyzer/WoWAnalyzer/blob/ab7815a8394297bb51b76e225692fdc97f43d6c5/src/analysis/retail/warrior/protection/normalizers/RageGenerationNormalizer.ts), which uses the same principle, but also trys to attribute various bonuses as a single normalizer, which becomes complicated, so I've separated the addition of new events with splitting/attribution into separate normalizers. Following the same model, attributing the rage gain from [Finishing Blows](https://www.wowhead.com/spell=400205/finishing-blows) to the talent Finishing Blows, rather than the talent [Strength of Arms](https://www.wowhead.com/spell=400803/strength-of-arms). Strength of Arms still exists, but does not have any rage gain anymore, so any logging of rage gains to that talent is just incorrect. 

If they were to adjust the logging event to correctly point to the Finishing Blows talent, it should work as expected, this normalizer will just be useless. If they were to change strength of arms to have some other rage gain in the future, it will be annoying and I'll have to figure out which rage is which, but it will require changes eitehr way.

I'd love to know if you think there's ways that are cleaner or more robust!

Reply:
i think that the splitting and reattributing in normalization is a pretty reasonable thing to do, especially when you have a lot of analyzers that rely on those numbers being correct. basically: that is doing hard, complicated things to make everything else easier down the line.

switching out a spell isn't really doing something hard. it makes some things a little bit easier, but isn't doing the same kind of heavy lifting.

however, there's a lot of churn happening right now and i also think it is important to let y'all get code online to support things now that the raid is out so i'm not gonna block the PR over this, just be mindful of it and if it ends up being easy to do this *without* rewriting the spell then maybe come back and adjust that

Reply:
@emallson I think that is a great way to view things. So would the other way to adjust the tracking of this spell in the ResourceTracker?